src/util/ajax.js

```diff
@@ -172,6 +176,7 @@ function makeFetchRequest(requestParameters: RequestParameters, callback: Respon
     }
 
     return { cancel: () => {
+        aborted = true;
         if (!complete) controller.abort();
```

Comment:
Can't the controller's abort signal be used directly to reject or escape from `validateOrFetch` early instead of using an additional `aborted` variable?

Reply:
from discussion in chat: let's leave this for now